### PR TITLE
formatting vstest output

### PR DESCRIPTION
**Pull request by prmittal**

### CODE REVIEW

**Comment by acesiddhu** [comment]
can u add more info. whats the bug and whats the fix

**Comment by prmittal** [comment]
The logs published by vstest to build do not differentiate between error and otherwise. Fix: take over handling of output events and print stderr as an error to build

**Comment by prmittal** [comment]
The logs published by vstest to build do not differentiate between error and otherwise. Fix: take over handling of output events and print stderr as an error to build

---
In reply to: [229677499](https://github.com/Microsoft/vsts-tasks/pull/1978#issuecomment-229677499) [](ancestors = 229677499)

File: Tasks/VsTest/vstest.ts [hunk 1]
```diff
@@ -202,7 +202,25 @@ function executeVstest(testResultsDirectory: string, parallelRunSettingsFile: st
     tl.rmRF(testResultsDirectory, true);
     tl.mkdirP(testResultsDirectory);
     tl.cd(workingDirectory);
-    vstest.exec()
+
+    // listener for standard output
+    vstest.on('stdline', function (line) {
+        tl._writeLine(line)
+    });
```
Comment (acesiddhu) [inline Tasks/VsTest/vstest.ts]: this should be fixed in task lib.

**Comment by prmittal** [comment]
These changes will cause error output from vstest to be pushed to stderr of the process. Once issue https://github.com/Microsoft/vsts-task-lib/issues/97 is fixed, the errors should show up in correct format.

**Comment by acesiddhu** [comment]
:shipit:

**Comment by nigurr** [comment]
Approved

**Pull request merged**